BaseClasses.py

```diff
@@ -586,7 +590,8 @@ def all_done():
 
             for location in sphere:
                 locations.remove(location)
```

Comment:
with locations being a list now, this is an order of magnitude slower

Reply:
the change i pushed increases performance beyond set's performance. it's a bit hard to measure because it has to finish before switching threads (waiting for the GIL). but the "best of" time is 10% faster than the set implementation.

Reply:
could test clearer by queuing this task first and then waiting for it to finish.

Reply:
I added a sleep(1) to avoid the task switching.

| | min | avg | max
| - | - | - | - |
| set (original) | 0.170 | 0.218 | 0.330
| list (this PR) | 0.151 | 0.198 | 0.339

of 20 runs with 10k locations